### PR TITLE
[Bug-685] map language is not the same as app language

### LINKED ISSUE #685: [Bug] Map language is not the same as app language
**Describe the bug**

**To Reproduce**
1. Go to home page
2. See default map language (herbrew)
3. change language to english
4. See - map language is still hebrew

**Expected behavior**
When passing params to google maps, we should pass language arguments as well to determined the proper language (should be the same as app language)

**Screenshots**
image

**Environment**
Browser: Chrome 
Environment:  Staging

### PR DESCRIPTION
**Pull request by daniel-ilan**
closes #685

### CODE REVIEW

File: src/components/molecules/map/GoogleMapsLayer.tsx [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 import { MAP_STYLE } from 'style/map.style';
 import { mapApiKey } from 'utils/utils';
 
+interface lng {
+  lng: string;
+}
+
 const GoogleMapsLayer: FC = ({ children }) => {
+  const { lng }: lng = useParams();
+  const selectedLang: string = lng ? lng : 'he';
```
Comment (almogbhl) [inline src/components/molecules/map/GoogleMapsLayer.tsx:13]: can we use `lng || 'he'` instead?
should we prefer to import a constant of `locales.hebrew`, is it the convention in the app or should we create one?

File: src/components/atoms/Link.tsx [hunk 2]
```diff
@@ -12,12 +12,23 @@ const useStyles = makeStyles({
...
+  shouldReload?: boolean;
 }
 
 const Link: FC<IProps> = ({ ...props }) => {
   const classes = useStyles();
+  const { shouldReload, to } = props;
+
+  if (shouldReload) {
+    return (
+      <a className={classes.link} {...props} href={to}>
```
Comment (almogbhl) [inline src/components/atoms/Link.tsx:27]: Do we want it to open the link in a new window so it opens the app again?
Can we use the store for renders and reload the relevant component?

Comment (daniel-ilan) [reply src/components/atoms/Link.tsx:27]: Using an anchor tag is not correct here, and was changed to a new variant of Button. 
We want the page to reload as we can't just rerender components. There is an issue in google maps api which doesn't allow changing the language on the fly. It's described in the issue #767

File: src/components/molecules/map/GoogleMapsLayer.tsx (hunk 1 above)
Comment (daniel-ilan) [reply src/components/molecules/map/GoogleMapsLayer.tsx:13]: I edited the code as you suggested, thanks.
Not sure about the `locals`, I need someone to tell me what to do about this @yuvalbl @SejoB

File: src/components/molecules/map/GoogleMapsLayer.tsx [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+import { useParams } from 'react-router-dom';
 import { MAP_STYLE } from 'style/map.style';
 import { mapApiKey } from 'utils/utils';
 
+interface lng {
+  lng: string;
+}
+
 const GoogleMapsLayer: FC = ({ children }) => {
+  const { lng }: lng = useParams();
```
Comment (yuvalbl) [inline src/components/molecules/map/GoogleMapsLayer.tsx:12]:
```suggestion
  const { lng } = useParams<lParam>();
```

Comment (yuvalbl) [reply src/components/molecules/map/GoogleMapsLayer.tsx:12]: see https://stackoverflow.com/questions/64517983/useparams-in-typescript-does-not-allow-destructuring

File: src/components/organisms/LanguageMenu.tsx [hunk 4]
```diff
@@ -28,6 +28,12 @@ const LanguageMenu: FC = () => {
   const closeMenu = () => {
     setAnchorEl(null);
   };
+  const reloadLang = (value: string) => {
+    if (value === 'he') {
+      return window.location.assign(`/newsflash/${store.activeNewsFlashId}`);
+    }
+    return window.location.assign(`/${value}/newsflash/${store.activeNewsFlashId}`);
+  };
```
Comment (yuvalbl) [inline src/components/organisms/LanguageMenu.tsx:36]:
```suggestion
  const LangClickHangler = (lang: string) => {
    const prefix = lang !== DEFAULT_LANG ? `/${lang}` : '';
    return window.location.assign(`${prefix}/newsflash/${store.activeNewsFlashId}`);
  };
```

File: src/components/molecules/map/GoogleMapsLayer.tsx [hunk 5]
```diff
@@ -1,12 +1,20 @@
...
+import { useParams } from 'react-router-dom';
 import { MAP_STYLE } from 'style/map.style';
 import { mapApiKey } from 'utils/utils';
 
+interface lng {
+  lng: string;
+}
+
 const GoogleMapsLayer: FC = ({ children }) => {
+  const { lng } = useParams<lng>();
```
Comment (yuvalbl) [inline src/components/molecules/map/GoogleMapsLayer.tsx:12]: can also be inline
```
useParams<{ lng: string}>();
```

File: src/components/molecules/map/GoogleMapsLayer.tsx [hunk 6]
```diff
@@ -1,12 +1,20 @@
...
+  const selectedLang: string = lng || 'he';
```
Comment (yuvalbl) [inline src/components/molecules/map/GoogleMapsLayer.tsx:13]:
```suggestion
  const language = lng || 'he';
```

File: src/components/molecules/map/GoogleMapsLayer.tsx [hunk 7]
```diff
@@ -1,12 +1,20 @@
...
+
   return (
     <ReactLeafletGoogleLayer
-      googleMapsLoaderConf={{ apiKey: mapApiKey, version: '3.40.6' }}
+      googleMapsLoaderConf={{ apiKey: mapApiKey, version: '3.40.6', language: selectedLang }}
```
Comment (yuvalbl) [inline src/components/molecules/map/GoogleMapsLayer.tsx:17]:
```suggestion
      googleMapsLoaderConf={{ apiKey: mapApiKey, version: '3.40.6', language }}
```

File: src/components/organisms/LanguageMenu.tsx [hunk 8]
```diff
@@ -22,12 +23,19 @@ const LANGUAGES = [
...
 const LanguageMenu: FC = () => {
   const [anchorEl, setAnchorEl] = useState(null);
+  const { i18n } = useTranslation();
   const openMenu = (event: any) => {
     setAnchorEl(event.currentTarget);
   };
   const closeMenu = () => {
     setAnchorEl(null);
   };
+  const LangClickHangler = (lang: string) => {
```
Comment (yuvalbl) [inline src/components/organisms/LanguageMenu.tsx:33]:
```suggestion
  const LangClickHandler = (lang: string) => {
```

File: src/components/organisms/LanguageMenu.tsx [hunk 9]
```diff
@@ -22,12 +23,19 @@ const LANGUAGES = [
...
+    if (lang !== i18n.language) {
+      const prefix = lang !== 'he' ? `/${lang}` : '';
+      return window.location.assign(`${prefix}/newsflash/${store.activeNewsFlashId}`);
```
Comment (yuvalbl) [inline src/components/organisms/LanguageMenu.tsx:36]:
```suggestion
      const prefix = lang !==  DEFAULT_LANGUAGE? `/${lang}` : '';
      return window.location.assign(`${prefix}/newsflash/${store.activeNewsFlashId}`);
```

**Review by yuvalbl — APPROVED** [review]

**Pull request merged**